feat: watchOS color palette implementation

billzhou0223 opened the pull request:
1. FioriThemeManager supports both iOS and WatchOS.
2. Created a new dedicated color palette `v1` for watchOS.
3. Setup watch test app.

MarcoEidinger reviewed (changes requested):
compiling `FioriThemeManager` for watchOS succeeds but running the related tests fail

<img width="1467" alt="Screenshot 2023-02-24 at 5 57 01 AM" src="https://user-images.githubusercontent.com/4176826/221096060-8b8fc986-4672-45bd-b6da-57ea933d773b.png">

I assume you forgot to make those tests unavailable for watchOS and rather run a new test to use/verify WatchPaletteV1

billzhou0223 commented:
> compiling `FioriThemeManager` for watchOS succeeds but running the related tests fail
> 
> <img alt="Screenshot 2023-02-24 at 5 57 01 AM" width="1467" src="https://user-images.githubusercontent.com/4176826/221096060-8b8fc986-4672-45bd-b6da-57ea933d773b.png">
> 
> I assume you forgot to make those tests unavailable for watchOS and rather run a new test to use/verify WatchPaletteV1

Fixed unit test issues. Also added availability check for ColorStyles. Please have another review.

MarcoEidinger reviewed (approved):
Looks good :)

Please merge (use squash), and don't worry that you have to bypass branch protections (as you can fix this by squashing and giving the final commit an appropriate message conforming to conventional commit guidelines)

After you merge this PR I will create another PR to change CI job to ensure that watch tests for FioriThemeManager will be executed as well.
